common.go

```diff
@@ -17,3 +18,20 @@ func debugLog(format string, args ...interface{}) {
 	}
 	log.Printf(format+"\n", args...)
 }
+
+//logURL replaces existing password from the given URL
+func logURL(URL string) (string, error) {
+	// Hide password if set in URL
+	u, err := url.Parse(URL)
+	if err != nil {
+		return "", err
```

Comment:
Maybe just return "" and avoid returning the error, which is just going to be discarded. That makes this function best-effort, which is fine.